nixos/lib/systemd-lib.nix

```diff
@@ -20,7 +20,10 @@ in rec {
       pkgs.runCommand "unit-${mkPathSafeName name}"
         { preferLocalBuild = true;
           allowSubstitutes = false;
-          inherit (unit) text;
+          # unit.text can be null. But variables that are null listed in
+          # passAsFile are ignored by nix, resulting in no file being created,
+          # making the mv operation fail.
+          text = unit.text ? "";
```

Comment:
Unless I am missing something obvious here, `?` is not a nullish coalescing operator (like `??` in JavaScript), but it is the "has attribute" operator. This would check if the attribute `""` is in `unit.text`, which will always return false.

Reply:
Yes, you are very much right.

Reply:
Thanks! Very good catch!

I somehow managed to completely look over that -- embarrassing.

I must have thought of default attribute values in function declarations or `or`.
But those two wouldn't work here anyway, because `text` is set, even if its value is `null`.

Oh well...

I should go to bed now, brain no longer work-y work-y 🫠